unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -285,13 +285,13 @@ public UTF8String trimRight() {
   }
 
   public UTF8String reverse() {
-    byte[] bytes = getBytes();
```

Comment:
This is the follow up for #6762, we'd better use the `copyMemory` instead of the `System.arrayCopy`, the later probably required additional `byte[]` created before the real copying.